src/Directory.Build.targets

```diff
@@ -1,4 +1,7 @@
 <Project>
+  <PropertyGroup Condition="'$(MinVerTagPrefix)' == 'core-'">
+    <PackageValidationBaselineVersion>$(OTelLatestStableVer)</PackageValidationBaselineVersion>
+  </PropertyGroup>
```

Comment:
Is there a reason this went into this file and not `Common.prod.props`? Most of the settings are there so I would like to move it if possible.

Reply:
Thanks for the comment! If I place this in `Common.prod.props` the package validation will simply not run. That's why it is here.

I believe this is because props are imported earlier than targets. Therefore, when the `PropertyGroup` is imported and used by the PackageValidation target, both `MinVerTagPrefix` and `OTelLatestStableVer` are empty. 

@ViktorHofer feel free to expand on this if you have any additional insights or suggestion.

Reply:
@mfogliatto Check this: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4923/files#diff-5baf5f9e448ad54ab25a091adee0da05d4d228481c9200518fcb1b53a65d4156R7-R12

Reply:
Just to follow up on this, yes there is an ordering issue. The `.props` files in the repo apply before the `csproj` so `MinVerTagPrefix` is not available to them. I just learned `msbuild` has a preprocess command you can use to spit out an XML to see everything in order:

`msbuild /pp:preprocess.xml src/OpenTelemetry/OpenTelemetry.csproj`

Kind of neat!

To get around that I just defined it as a target so we can control when it runs/applies the properties.

Reply:
> @mfogliatto Check this: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4923/files#diff-5baf5f9e448ad54ab25a091adee0da05d4d228481c9200518fcb1b53a65d4156R7-R12

Note that this might not work with static graph restore (which is a NuGet feature). Those properties are kind of expected to be set at evaluation time, not inside a target. IMO where they are right now is fine.